torch/csrc/autograd/python_legacy_variable.cpp

```diff
@@ -0,0 +1,126 @@
+#include "python_variable.h"
+
+#include <ATen/ATen.h>
+
+#include "torch/csrc/Exceptions.h"
+#include "torch/csrc/autograd/python_function.h"
+#include "torch/csrc/autograd/python_variable.h"
+#include "torch/csrc/tensor/python_tensor.h"
+
+using namespace at;
+
+namespace torch { namespace autograd {
+
+static PyObject *THPVariable_pynew(PyTypeObject* type, PyObject *args, PyObject *kwds) {
```

Comment:
We keep this in C++ only for perf reasons, right?

Reply:
Perf was the main reason I kept this as C++. I'm also not sure we expose enough of the API to be able to fully re-implement this in Python. (For example, I'm not sure we expose `set_name`)